src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Workspace/Mef/ILayeredServiceMetadata.cs

```diff
@@ -8,7 +8,7 @@ namespace Microsoft.CodeAnalysis.Host.Mef;
 
 internal interface ILayeredServiceMetadata
 {
-    public IReadOnlyList<string> WorkspaceKinds { get; }
+    public WorkspaceKinds WorkspaceKinds { get; }
```

Comment:
Anybody using us a library can make up their own kinds, won't this break them?

Reply:
Ah I see this is an internal filter, but we _also_ use the Layer as a workspace kind as well.

Reply:
Wo don't use layer as workspace kind. I cleaned that up. We use either layer or string[] of workspace kinds. The latter is now enum.